### PR TITLE
refactor: remove macros for nodejs_binary and nodejs_test

### PR DESCRIPTION
**Pull request by meteorcloudy**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by meteorcloudy** [comment]
@alexeagle Can you help with the failure of `//packages/protractor/test/protractor-2:prodserver_test_chromium-local`, I don't quite understand what's happening.

**Comment by meteorcloudy** [comment]
@alexeagle Tests are all green, please take a look!

**Comment by meteorcloudy** [comment]
I'll be on vacation for the next three weeks, so feel free to make any changes to this pr directly ;)

**Comment by meteorcloudy** [comment]
Oh, looks like we need one more fix from https://github.com/angular/angular-bazel-example/blob/9efe352a84ccb50a798ea6873ec5c349d7604691/e2e/protractor.on-prepare.js#L17 for the e2e_angular_bazel_example test, @alexeagle can you help with this?

**Comment by alexeagle** [comment]
@meteorcloudy you are my hero :rose:

@gregmagolan can help with that last failure, he was just working in the protractor stuff in angular-bazel-example so he has the context

Should add to the commit when we squash:
fixes #635 
fixes #51

**Comment by gregmagolan** [comment]
@googlebot I consent

File: internal/bazel_integration_test/git_repository_under_test.bzl [hunk 1]
```diff
@@ -69,6 +69,41 @@ _attrs = {
         default = False,
         doc = "Whether to clone submodules in the repository.",
     ),
+    "patch_args": attr.string_list(
```
Comment (alexeagle) [inline internal/bazel_integration_test/git_repository_under_test.bzl]: can't this be a `**kwargs` to repository_rule or whatever you're wrapping, so we don't have to declare/document so many attributes?

Comment (gregmagolan) [reply internal/bazel_integration_test/git_repository_under_test.bzl]: moved to https://github.com/bazelbuild/rules_nodejs/pull/1105; in this case I am duplicating the implementation of the git_repository repository rule here but I use the vendored utils.bzl for most of the work so its mostly a shell implementation so have to define these attributes here

**Pull request merged**